Maple2.File.Parser/AiParser.cs

```diff
@@ -10,24 +10,265 @@
 using Maple2.File.Parser.Tools;
 using Maple2.File.Parser.Xml.Table;
 using System.Xml;
+using System.Text;
+using System.Xml.Linq;
+using System;
+using System.Reflection;
+using System.Numerics;
+using Maple2.File.Parser.Flat.Convert;
 
 namespace Maple2.File.Parser;
 
 public class AiParser {
     private readonly M2dReader xmlReader;
-    private readonly XmlSerializer aiSerializer;
 
     public AiParser(M2dReader xmlReader) {
         this.xmlReader = xmlReader;
-        this.aiSerializer = new XmlSerializer(typeof(NpcAi));
+    }
+
+    public void ParseAttributes<T>(XmlNode xmlNode, T outObject) {
+        foreach (XmlAttribute attrib in xmlNode.Attributes) { 
+            FieldInfo? field = typeof(T).GetField(attrib.Name);
+
+            if (field is null)
+            {
+                throw new MemberAccessException($"object of type '{typeof(T).Name}' doesn't have member '{attrib.Name}' from node '{xmlNode.LocalName}'");
+            }
+
+            switch (field.FieldType.Name)
+            {
+                case "Boolean":
+                    if (!bool.TryParse(attrib.Value, out bool outBool)) {
+						if (!int.TryParse(attrib.Value, out int outBoolInt)) {
+                        	throw new InvalidDataException($"invalid bool '{attrib.Value}' for {typeof(T).Name} member {field.Name}");
+						}
+
+						outBool = outBoolInt != 0;
+                    }
+
+                    field.SetValue(outObject, outBool);
+                    break;
+                case "Int16":
+                    if (!short.TryParse(attrib.Value, out short outShort))
+                    {
+                        throw new InvalidDataException($"invalid bool '{attrib.Value}' for {typeof(T).Name} member {field.Name}");
```

Comment:
All these error messages say "bool" but they are not bool

Reply:
lol woops, fixing